src/main/java/de/tum/in/www1/artemis/repository/StatisticsRepository.java

```diff
@@ -0,0 +1,109 @@
+package de.tum.in.www1.artemis.repository;
+
+import java.time.Instant;
+import java.time.ZonedDateTime;
+import java.util.List;
+import java.util.Map;
+
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
+import org.springframework.stereotype.Repository;
+
+import de.tum.in.www1.artemis.domain.User;
+
+/**
+ * Spring Data JPA repository for the user statistics
+ */
+@Repository
+public interface StatisticsRepository extends JpaRepository<User, Long> {
+
+    @Query("""
+            select count(distinct u.login)
+            from User u, PersistentAuditEvent p
+            where u.login like p.principal and p.auditEventType = 'AUTHENTICATION_SUCCESS' and u.login not like '%test%' and p.auditEventDate >= :#{#span}
+            """)
+    Integer getLoggedInUsers(@Param("span") Instant span);
+
+    @Query("""
+            select count(distinct u.login)
+            from User u, Submission s, StudentParticipation p
+            where s.participation.id = p.id and p.student.id = u.id and s.submissionDate >= :#{#span} and u.login not like '%test%'
+            """)
+    Integer getActiveUsers(@Param("span") ZonedDateTime span);
```

Comment:
Might be better to rename this to something like `getActiveUserCountFromSpan`?

Reply:
Or maybe "active" is the wrong word here since the query returns users that have submitted something from a given date.

Reply:
This deprecated code, i really should have marked it somehow, sorry. I will refactor most of the server side code.